apps/web/src/components/common/sidebar/sections/projects/create-project-modal.tsx

```diff
@@ -57,8 +58,32 @@ function CreateProjectModal({ open, onClose }: CreateProjectModalProps) {
     setSlug(generateProjectSlug(newName));
   };
 
+  const resetProjectState = async () => {
+    await queryClient.invalidateQueries({ queryKey: ["projects"] });
+    resetState();
+  };
+
+  const resetState = () => {
+    setName("");
+    setSlug("");
+    setSelectedIcon("Layout");
+  };
+
+  const resetAndCloseModal = () => {
+    resetProjectState();
+    onClose();
+  };
+
+  useEffect(() => {
+    if (open) {
+      requestAnimationFrame(() => {
+        projectNameRef.current?.focus();
+      });
+    }
+  }, [open]);
```

Comment:
suggestion: I believe we can remove the ref and useEffect if we set an `autoFocus` prop to the input element itself.